### PR TITLE
Implement Online Datasource [specific ci=Group1-Docker-Commands]

### PR DESCRIPTION
**Pull request by morris-jason**
Implements Online Docker CP, including ContainerStatPath, using
Guest Tools and the Toolbox. These changes also introduce storage
handlers for Docker CP implementations of Export/Import Archive
and personality code for ContainerArchivePath and
ContainerExtractToDir. Fixes 5606, 5715,  and 5570.

### CODE REVIEW

File: lib/apiservers/engine/backends/container_proxy.go [hunk 1]
```diff
@@ -153,6 +151,7 @@ const (
 	swaggerSubstringEOF                 = "EOF"
 	forceLogType                        = "json-file" //Use in inspect to allow docker logs to work
 	ShortIDLen                          = 12
+	archiveStreamBufSize                = 64 * 1024
```
Comment (morris-jason) [inline lib/apiservers/engine/backends/container_proxy.go]: not sure what this came from. will delete.

File: lib/archive/diff.go [hunk 2]
```diff
@@ -51,19 +51,18 @@ func Diff(op trace.Operation, newDir, oldDir string, spec *FilterSpec, data bool
...
 	if err != nil {
 		return nil, err
 	}
-	// this is in the case that we are part of a read that is crossing a mount point.
-	// It should be addressed by include/excludes.
-	changes = append(changes, docker.Change{Path: "/"})
+	changes = append(changes, dockerChanges...)
 
 	sort.Sort(changesByPath(changes))
-	op.Infof("FILESPEC:::: \n\n\n %#v", *spec)
```
Comment (matthewavery) [inline lib/archive/diff.go]: Ha thanks for removing this, really thought i had removed all of those...

File: lib/archive/unpack.go [hunk 3]
```diff
@@ -91,124 +94,34 @@ func Unpack(op trace.Operation, tarStream io.Reader, filter *FilterSpec, root st
...
 			}
-
 		default:
 			// TODO: add support for special file types - otherwise we will do absurd things such as read infinitely from /dev/random
 		}
 	}
 	return nil
 }
-
-func processFirstEntry(op trace.Operation, tarStream *tar.Reader, filter *FilterSpec, root string) error {
```
Comment (matthewavery) [inline lib/archive/unpack.go]: yeah with ContainerStatPath running well there is no longer a need for this hack :)

File: pkg/vsphere/disk/disk_manager.go [hunk 4]
```diff
@@ -538,13 +538,19 @@ func (m *Manager) UnmountAndDetach(op trace.Operation, datastoreURI *object.Data
 func (m *Manager) InUse(op trace.Operation, config *VirtualDiskConfig, filter func(vm *mo.VirtualMachine) bool) ([]*vm.VirtualMachine, error) {
 	defer trace.End(trace.Begin(""))
 
-	if m.view == nil {
-		return nil, fmt.Errorf("ContainerView is nil")
+	mngr := view.NewManager(m.vm.Vim25())
+
+	// Create view of VirtualMachine objects under the VCH's resource pool
+	view2, err := mngr.CreateContainerView(op, m.vm.Session.Pool.Reference(), []string{"VirtualMachine"}, true)
```
Comment (matthewavery) [inline pkg/vsphere/disk/disk_manager.go]: why view2 and not `view`?

**Comment by matthewavery** [comment]
Some minor comments, you can merge away from here. Then we can begin resolving all the unit tests failures.

**Pull request merged**